src/smart-components/role/edit-resource-definitions-modal.js

```diff
@@ -0,0 +1,193 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import componentTypes from '@data-driven-forms/react-form-renderer/dist/cjs/component-types';
+import FormRenderer from '../common/form-renderer';
+import { useHistory, useRouteMatch } from 'react-router-dom';
+import { useDispatch, useSelector, shallowEqual } from 'react-redux';
+import { updateRole, fetchRole } from '../../redux/actions/role-actions';
+import { routes as paths } from '../../../package.json';
+import { getResource, getResourceDefinitions } from '../../redux/actions/cost-management-actions';
+import componentMapper from '@data-driven-forms/pf4-component-mapper/dist/cjs/component-mapper';
+import { WarningModal } from '../common/warningModal';
+import { Spinner, Modal } from '@patternfly/react-core';
+import ResourceDefinitionsFormTemplate from './ResourceDefinitionsFormTemplate';
+import './role-permissions.scss';
+
+const createOptions = (resources) => {
+  let options = [];
+  for (const [key, value] of Object.entries(resources)) {
```

Comment:
I think we should avoid using for loops unless there is specific reason to use it.
```
Object.entries(resources).reduce((acc, [key, value]) => [...acc, value.map((r) => ({ value: r.value, path: key, label: r.value }))], []);
```

Reply:
Actually when you make changes to the selector, this can be:
```
const createOptions = (resources) => resources.map((r) => ({ value: r.value, label: r.value }));
```